appservice/src/createAppService/AppInsightsListStep.ts

```diff
@@ -0,0 +1,85 @@
+import { AppInsightsCreateStep } from "./AppInsightsCreateStep";
+import { AppInsightsLocationStep } from "./AppInsightsLocationStep";
+import { AppInsightsNameStep } from "./AppInsightsNameStep";
+import { IAppServiceWizardContext } from "./IAppServiceWizardContext";
+
+export const appInsightsNamingRules: IAzureNamingRules = {
+    minLength: 1,
+    maxLength: 255,
+    invalidCharsRegExp: /[^a-zA-Z0-9\.\_\-\(\)]/
+};
+
+const skipForNow: string = '$(circle-slash) Skip for now';
+
+export class AppInsightsListStep extends AzureWizardPromptStep<IAppServiceWizardContext> {
+    public static async getAppInsightsComponents(wizardContext: IAppServiceWizardContext): Promise<ApplicationInsightsComponentListResult> {
+        const client: ApplicationInsightsManagementClient = createAzureClient(wizardContext, ApplicationInsightsManagementClient);
+        return await client.components.list();
+    }
+
+    public static async isNameAvailable(wizardContext: IAppServiceWizardContext, name: string): Promise<boolean> {
+        const resourceGroupsTask: Promise<ApplicationInsightsComponentListResult> = AppInsightsListStep.getAppInsightsComponents(wizardContext);
+        return !(await resourceGroupsTask).some((rg: ApplicationInsightsComponent) => rg.name !== undefined && rg.name.toLowerCase() === name.toLowerCase());
+    }
+
+    public async prompt(wizardContext: IAppServiceWizardContext): Promise<void> {
+        const options: IAzureQuickPickOptions = { placeHolder: 'Select an App Insights component for your app.', id: `AppInsightsListStep/${wizardContext.subscriptionId}` };
+        wizardContext.appInsightsComponent = (await ext.ui.showQuickPick(this.getQuickPicks(wizardContext), options)).data;
+        if (wizardContext.appInsightsComponent) {
+            if (isString(wizardContext.appInsightsComponent)) {
+                wizardContext.telemetry.properties.skipForNow = 'true';
+            }
+        }
+    }
+
+    public shouldPrompt(wizardContext: IAppServiceWizardContext): boolean {
+        return !wizardContext.appInsightsComponent && !!wizardContext.location;
+    }
+
+    public async getSubWizard(wizardContext: IAppServiceWizardContext): Promise<IWizardOptions<IAppServiceWizardContext> | undefined> {
+        if (!wizardContext.appInsightsComponent && wizardContext.appInsightsComponent !== skipForNow) {
+            return {
+                promptSteps: [new AppInsightsNameStep(), new AppInsightsLocationStep()],
+                executeSteps: [new AppInsightsCreateStep()]
+            };
+        } else {
+            return undefined;
+        }
+    }
+
+    private async getQuickPicks(wizardContext: IAppServiceWizardContext): Promise<IAzureQuickPickItem<ApplicationInsightsComponent | string | undefined>[]> {
+
+        const picks: IAzureQuickPickItem<ApplicationInsightsComponent | string | undefined>[] = [{
+            label: localize('newApplicationInsight', '$(plus) Create new application insight'),
```

Comment:
The portal tends to use Application Insight and then resource name.  I used the term component because that's how the API refers to them, but I actually don't see that vernacular anywhere in the portal, unfortunately.  I'd guess we should probably call it an `Application Insight resource`

![image](https://user-images.githubusercontent.com/5290572/62975563-8512ea00-bdcf-11e9-9b1a-470d907fc834.png)

![image](https://user-images.githubusercontent.com/5290572/62975589-94923300-bdcf-11e9-9db3-ad17fe876c57.png)